asf_tools/composite.py

```diff
@@ -0,0 +1,308 @@
+#!/usr/bin/env python
+"""Create S1 SAR Composite Mosaic using inverse area weighting ala David Small.
+
+   Path vs infiles:
+     If path is passed, code assumes files are in an ASF HyP3 RTC Stacking arrangement.
+     i.e  {path}/20*/PRODUCT/ contains the input RTC data and the area maps or
+          {path}/S1?_IW_*RTC*/ contains the input RTC data and the area maps
+         
+
+"""
+
+import numpy as np
+import logging
+import os
+import re
+import argparse
+import glob
+# from hyp3lib import saa_func_lib as saa
+# import hyp3lib.saa_func_lib as saa
+import saa_func_lib as saa
+from osgeo import gdal
```

Comment:
We should drop this as a requirement so we don't depend on `hyp3lib`

Reply:
I don't understand.  What's the point of having hyp3lib if we can't/don't call it from our codes?
I thought that was the point, to allow reuse of code in a convenient library.

Reply:
As we've discussed, `hyp3lib` should be restricted to hyp3-y stuff and using it here pulls in a lot of dependencies that aren't needed. We'll be pulling general tools out of hyp3lib and distributing them here -- we've always discussed this being a general tool, not specific to HyP3. 

It's a lot easier to develop and manage when packages have a defined scope -- GAMMA stuff to hyp3-gamma, ISCE stuff to hyp3-isce, general tools here, and general hyp3 stuff in hyp3lib. Using pip/conda means it's just an install command to get whatever combination you need.

Reply:
So, its just hyp3lib that you are opposed to using because you want it only for hyp3 stuff? That makes sense.  But, then where do I put functions like this in saa_func_lib?  They don't really go with asf geometry.

Reply:
For this package we can implement any needed functionality into  appropriately named modules/functions. `saa_func_lib` is often doing more work than is needed (e.g., only using one of many returns, reading the whole file when not needed) based on older gdal interfaces/limitation, or is direct passthroughs to gdal functions.